### PR TITLE
Fix task summary doctest 

### PR DESCRIPTION
**Pull request by stevhliu**
The doctests are failing for the updated task summary page because outputs weren't included in the code examples. This PR adds real inputs that can be pipelined instead of the generic `("path/to/data/file")`. I skipped the `text-generation` pipeline, but let me know if you'd prefer setting a seed for it so we can still reliably generate an output.

### CODE REVIEW

File: docs/source/en/task_summary.mdx [hunk 1]
```diff
@@ -34,8 +34,12 @@ Audio classification is a task that labels audio data from a predefined set of c
...
 >>> from transformers import pipeline
 
->>> classifier = pipeline(task="audio-classification")
->>> classifier("path/to/audio/file.mp3")
+>>> classifier = pipeline(task="audio-classification", model="superb/hubert-base-superb-er")
+>>> classifier("https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/mlk.flac")
+[{'score': 0.453214168548584, 'label': 'hap'},
+ {'score': 0.3622128367424011, 'label': 'sad'},
+ {'score': 0.09430035948753357, 'label': 'neu'},
+ {'score': 0.09027273952960968, 'label': 'ang'}]
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:42]: Scores should only be printed with 3 or 4 digits, or the tests can't be reproduced on different hardware.

File: docs/source/en/task_summary.mdx [hunk 2]
```diff
@@ -73,7 +78,14 @@ Image classification labels an entire image from a predefined set of classes. Li
...
 >>> classifier = pipeline(task="image-classification")
->>> classifier("path/to/image/file.jpg")
+>>> classifier("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg")
+[{'score': 0.44030314683914185, 'label': 'lynx, catamount'},
+ {'score': 0.034334052354097366,
+  'label': 'cougar, puma, catamount, mountain lion, painter, panther, Felis concolor'},
+ {'score': 0.03214803338050842,
+  'label': 'snow leopard, ounce, Panthera uncia'},
+ {'score': 0.02353905513882637, 'label': 'Egyptian cat'},
+ {'score': 0.02303415723145008, 'label': 'tiger cat'}]
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:88]: Same here.

File: docs/source/en/task_summary.mdx [hunk 3]
```diff
@@ -88,7 +100,10 @@ Unlike image classification, object detection identifies multiple objects within
 >>> from transformers import pipeline
 
 >>> detector = pipeline(task="object-detection")
->>> detector("path/to/image/file.jpg")
+>>> detector("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg")
+[{'score': 0.9865359663963318,
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:104]: And there

File: docs/source/en/task_summary.mdx [hunk 4]
```diff
@@ -104,7 +119,16 @@ Segmentation tasks are helpful in self-driving vehicles to create a pixel-level
...
+>>> segmenter("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg")
+[{'score': 0.985624,
+  'label': 'LABEL_184',
+  'mask': <PIL.Image.Image image mode=L size=960x686 at 0x7F86A0511670>},
+ {'score': 0.997607,
+  'label': 'snow',
+  'mask': <PIL.Image.Image image mode=L size=960x686 at 0x7F869FA06E50>},
+ {'score': 0.996198,
+  'label': 'cat',
+  'mask': <PIL.Image.Image image mode=L size=960x686 at 0x7F869FA06EE0>}]
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:131]: Here too.

File: docs/source/en/task_summary.mdx [hunk 5]
```diff
@@ -120,7 +144,17 @@ There are two approaches to depth estimation:
 >>> from transformers import pipeline
 
 >>> depth_estimator = pipeline(task="depth-estimation")
->>> depth_estimator("path/to/image/file.jpg")
+>>> depth_estimator(
+...     "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
+... )
+{'predicted_depth': tensor([[[ 0.7767,  0.8142,  0.8237,  ...,  2.3059,  2.3640,  2.3277],
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:150]: This doesn't really bring anything to the table, so let's not show the output here.

File: docs/source/en/task_summary.mdx [hunk 6]
```diff
@@ -155,6 +190,7 @@ Two common types of token classification are:
 
 >>> classifier = pipeline(task="ner")
 >>> classifier("Hugging Face is a French company based in New York City.")
+[{'entity': 'I-ORG', 'score': 0.9967675, 'index': 1, 'word': 'Hu', 'start': 0, 'end': 2}, {'entity': 'I-ORG', 'score': 0.9293027, 'index': 2, 'word': '##gging', 'start': 2, 'end': 7}, {'entity': 'I-ORG', 'score': 0.9763208, 'index': 3, 'word': 'Face', 'start': 8, 'end': 12}, {'entity': 'I-MISC', 'score': 0.99828726, 'index': 6, 'word': 'French', 'start': 18, 'end': 24}, {'entity': 'I-LOC', 'score': 0.99896204, 'index': 10, 'word': 'New', 'start': 42, 'end': 45}, {'entity': 'I-LOC', 'score': 0.9986792, 'index': 11, 'word': 'York', 'start': 46, 'end': 50}, {'entity': 'I-LOC', 'score': 0.9992418, 'index': 12, 'word': 'City', 'start': 51, 'end': 55}]
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:193]: All scores need to be shown with only 3 or 4 digits.

File: docs/source/en/task_summary.mdx [hunk 7]
```diff
@@ -139,6 +173,7 @@ Like classification tasks in any modality, text classification labels a sequence
 
 >>> classifier = pipeline(task="sentiment-analysis")
 >>> classifier("Hugging Face is the best thing since sliced bread!")
+[{'label': 'POSITIVE', 'score': 0.9990912675857544}]
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:176]: Same here.

File: docs/source/en/task_summary.mdx [hunk 8]
```diff
@@ -175,6 +211,7 @@ There are two common types of question answering:
 ...     question="What is the name of the repository?",
 ...     context="The name of the repository is huggingface/transformers",
 ... )
+{'score': 0.9327335953712463, 'start': 30, 'end': 54, 'answer': 'huggingface/transformers'}
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:214]: Same problem here.

File: docs/source/en/task_summary.mdx [hunk 9]
```diff
@@ -220,17 +259,21 @@ There are two types of language modeling:
...
 
 * masked: the model's objective is to predict a masked token in a sequence with full access to the tokens in the sequence
     
     ```py
-    >>> text = "Hugging Face is a <mask> company based in New York City."
+    >>> text = "Hugging Face is a community-based open-source <mask> for machine learning."
     >>> fill_mask = pipeline(task="fill-mask")
-    >>> fill_mask(text, top_k=3)
+    >>> fill_mask(text, top_k=1)
+    [{'score': 0.22355437278747559,
```
Comment (sgugger) [inline docs/source/en/task_summary.mdx:273]: And there.

**Review by sgugger — COMMENTED** [review]
Thanks for adding those.

As explained in the doc for doctests, the numbers printed should only use a small precision, as different hardwares will produce results that are only within 1e-5 of difference, not precisely equal. Also some of the outputs are not useful at all, so should be stored in a variable and not shown.

File: docs/source/en/task_summary.mdx [hunk 10]
```diff
@@ -73,7 +80,17 @@ Image classification labels an entire image from a predefined set of classes. Li
 >>> from transformers import pipeline
 
 >>> classifier = pipeline(task="image-classification")
->>> classifier("path/to/image/file.jpg")
+>>> preds = classifier(
+...     "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
+... )
+>>> preds = [{"score": round(pred["score"], 4), "label": pred["label"]} for pred in preds]
+>>> preds
+[{'score': 0.4403, 'label': 'lynx, catamount'},
```
Comment (ydshieh) [inline docs/source/en/task_summary.mdx:88]: Is this formatted by `make style`? I feel we can have a better representation. See comments below too.

File: docs/source/en/task_summary.mdx [hunk 11]
```diff
@@ -154,7 +196,61 @@ Two common types of token classification are:
...
+...         "score": round(pred["score"], 4),
+...         "index": pred["index"],
+...         "word": pred["word"],
+...         "start": pred["start"],
+...         "end": pred["end"],
+...     }
+...     for pred in preds
+... ]
+>>> preds
+[{'entity': 'I-ORG',
```
Comment (ydshieh) [inline docs/source/en/task_summary.mdx:212]: well, this is what I mean at the beginning of my review. It's better to have each entity (here a dictionary) in a single line. Let me know if you can make it :-)

File: docs/source/en/task_summary.mdx [hunk 12]
```diff
@@ -171,10 +267,31 @@ There are two common types of question answering:
...
+>>> preds = question_answerer(
 ...     question="What is the name of the repository?",
 ...     context="The name of the repository is huggingface/transformers",
 ... )
+>>> preds = [
+...     {"score": round(preds["score"], 4), "start": preds["start"], "end": preds["end"], "answer": preds["answer"]}
+...     for pred in preds
+... ]
+>>> preds
+[{'score': 0.9327,
```
Comment (ydshieh) [inline docs/source/en/task_summary.mdx:279]: same comment as above.

**Review by ydshieh — COMMENTED** [review]
Thank you @stevhliu for working on this. I think we can improve this PR better with some style change.

File: docs/source/en/task_summary.mdx (hunk 11 above)
Comment (stevhliu) [reply docs/source/en/task_summary.mdx]: Let me know if the new changes are better!

Comment (ydshieh) [reply docs/source/en/task_summary.mdx]: Thank you @stevhliu , definitely better (at least for me, hopefully for you too :-) ) ❤️ . 

I am still not a fan of
```python
[{'score': 0.9856, 'label': 'LABEL_184'},
 {'score': 0.9976, 'label': 'snow'},
 {'score': 0.9962, 'label': 'cat'}]
```
(I usually get used to a proper indentation), but I don't want to be picky. As long as the doctest pass and Sylvain is happy too, we can keep like this.

**Review by sgugger — APPROVED** [review]

**Review by ydshieh — APPROVED** [review]

File: docs/source/en/task_summary.mdx (hunk 11 above)
Comment (stevhliu) [reply docs/source/en/task_summary.mdx]: Definitely better for me too, thank you!

**Pull request merged**